stock_available/views/product_template_view.xml

```diff
@@ -76,9 +76,11 @@
         <field name="model">product.template</field>
         <field name="inherit_id" ref="stock.product_template_kanban_stock_view" />
         <field name="arch" type="xml">
-            <xpath expr="//div[@name='product_lst_price']" position="after">
+            <xpath expr="//span/field[@name='list_price']" position="after">
+                <field name="type" invisible="1" />
+                <field name="is_storable" invisible="1" />
                 <div
-                    t-if="record.type.raw_value == 'product'"
+                    t-if="record.type.raw_value == 'consu' and record.is_storable.raw_value"
```

Comment:
Analogous to https://github.com/odoo/odoo/blob/686628f55f6ec/addons/stock/views/product_views.xml#L179, you should add the group attribute, and you could choose to use `show_on_hand_qty_status_button` as a criteria. This attribute is computed as True for products that are is_storable, unless they are a kit product (as per mrp's override of the compute method). That way, the quantity fields are all consistently visible.